engine/src/conversion/analysis/remove_ignored.rs

```diff
@@ -69,13 +69,25 @@ pub(crate) fn filter_apis_by_ignored_dependents(mut apis: Vec<Api<FnPhase>>) ->
     apis
 }
 
-fn create_ignore_item(api: Api<FnPhase>, err: ConvertError) -> Api<FnPhase> {
+fn create_ignore_item(api: Api<FnPhase>, err: ConvertError) -> Option<Api<FnPhase>> {
     let id = api.name().get_final_ident();
     log::info!("Marking as ignored: {} because {}", id.to_string(), err);
-    Api::IgnoredItem {
+    Some(Api::IgnoredItem {
         name: api.name_info().clone(),
         err,
         ctx: match api {
+            Api::Function {
+                analysis:
+                    FnAnalysis {
+                        kind: FnKind::TraitMethod { .. },
+                        ..
+                    },
+                ..
+            } => {
+                // We're not going to generate the impl block for the trait, which means there's no
+                // place to attach an error.
+                return None;
```

Comment:
I can see why you've done this, I'm just a tiny bit worried about unintended consequences here. The dependency-tracking stuff is fragile, and I am worried that there might be cases where the existence of some kind of `Api` - even an `IgnoredItem` - is necessary for the graph of dependencies to work properly.

So, because I'm paranoid, please continue to generate an `IgnoredItem`, but add a new enum variant to `ErrorContext` which says `None` and represents the fact that we can't generate code to represent this error. This is hopefully a smaller code change too.

Reply:
Done